src/shiftml/ase/calculator.py

```diff
@@ -0,0 +1,143 @@
+import os
+import urllib.request
+
+from metatensor.torch.atomistic import ModelOutput
+from metatensor.torch.atomistic.ase_calculator import MetatensorCalculator
+
+url_resolve = {
+    "ShiftML1.0": "https://tinyurl.com/3xwec68f",
+    "ShiftML1.1": "https://tinyurl.com/53ymkhvd",
+}
+
+resolve_outputs = {
+    "ShiftML1.0": {"mtt::cs_iso": ModelOutput(quantity="", unit="ppm", per_atom=True)},
+    "ShiftML1.1": {"mtt::cs_iso": ModelOutput(quantity="", unit="ppm", per_atom=True)},
+}
+
+resolve_fitted_species = {
+    "ShiftML1.0": set([1, 6, 7, 8, 16]),
+    "ShiftML1.1": set([1, 6, 7, 8, 16]),
+}
+
+
+class ShiftML(MetatensorCalculator):
+    """
+    ShiftML calculator for ASE
+    """
+
+    def __init__(self, model_version, force_download=False):
+        """
+        Initialize the ShiftML calculator
+
+        Parameters
+        ----------
+        model_version : str
+            The version of the ShiftML model to use. Supported versions are
+            "ShiftML1.0".
+        force_download : bool, optional
+            If True, the model will be downloaded even if it is already in the cache.
+            Default is False.
+        """
+
+        try:
+            # The rascline import is necessary because
+            # it is required for the scripted model
+            import rascaline.torch
```

Comment:
you may also want to check that the installed rascaline version is what you want